lib/watson-discovery-setup.js

```diff
@@ -324,12 +323,44 @@ WatsonDiscoverySetup.prototype.setupDiscovery = function(setupParams, callback)
 };
 
 /**
- * Add files to Discovery collection.
+ * Add chunk of files to Discovery collection.
  */
 WatsonDiscoverySetup.prototype.loadDiscoveryData = function (collectionParams, callback) {
   this.loadDiscoveryCollection(collectionParams)
     .then(params => callback(null, params))
     .catch(callback);
 };
 
+/**
+ * Manage the flow of files being added to the Discovery collection.
+ */
+WatsonDiscoverySetup.prototype.loadCollectionFiles = function (params) {
+  this.loadDiscoveryData(params, (err, data) => {
+    if (err) {
+      handleSetupError(err);
+      console.log(err);
+    } else {
+      var collectionParams = data;
+      if ((! collectionParams.docsAlreadyLoaded) && (collectionParams.docCurrentIdx < collectionParams.numDocs)) {
+        this.loadCollectionFiles(collectionParams);
+      } else {
+        console.log('Discovery collection loading has completed!');
+      }
+    }
+  });
+};
+
+/**
+ * Handle setup errors by logging and appending to the global error text.
+ * @param {String} reason - The error message for the setup error.
+ */
+function handleSetupError(reason) {
+  setupError += ' ' + reason;
```

Comment:
setupError defined?  I was using a global here w/ that comment below that instead of exit it could let the chatbot report the error.  Maybe the exit below is OK but looking for unresolved setupError(?)